src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,256 @@
+        } catch (Exception x) {
+            logMessageProcessingFailureNonRecoverable(message, ORDER_RECEIVED_TOPIC, x);
+        }
     }
 
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC, groupId = ORDER_RECEIVED_GROUP,
-            autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
-    public void processOrderReceived(String orderReceivedUri)
-            throws SerializationException, ExecutionException, InterruptedException {
+    /**
+     * Retry (`-retry`) listener/consumer. Calls a `retryable` method to process received message.
+     * If the `retryable` processor is unsuccessful with a `retryable` error, after maximum numbers of attempts allowed,
+     * the message is published to `-error` topic for failover processing.
+     * @param message
+     * @throws SerializationException
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @KafkaListener(id = ORDER_RECEIVED_GROUP_RETRY, groupId = ORDER_RECEIVED_GROUP_RETRY,
+                    topics = ORDER_RECEIVED_TOPIC_RETRY,
+                    autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
+    public void processOrderReceivedRetry(org.springframework.messaging.Message<OrderReceived> message)
+            throws InterruptedException, ExecutionException, SerializationException {
+        OrderReceived orderReceived = message.getPayload();
+        String orderReceivedUri = orderReceived.getOrderUri();
         try {
-            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC);
-        } catch (Exception ex){
-            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY, ex);
+            logMessageReceived(message, ORDER_RECEIVED_TOPIC_RETRY);
+
+            logMessageProcessed(message, ORDER_RECEIVED_TOPIC_RETRY);
+        } catch (ServiceException ex){
+            logMessageProcessingFailureRecoverable(message, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR, ex);
+            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR);
+        } catch (Exception x) {
+            logMessageProcessingFailureNonRecoverable(message, ORDER_RECEIVED_TOPIC_RETRY, x);
         }
     }
 
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC_RETRY, groupId = ORDER_RECEIVED_GROUP_RETRY,
-            autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
-    public void processOrderReceivedRetry(String orderReceivedUri)
+    /**
+     * Error (`-error`) topic listener/consumer is enabled when the application is launched in error
+     * mode (IS_ERROR_QUEUE_CONSUMER=true). Receives messages up to `ERROR_RECOVERY_OFFSET` offset. Calls a `retryable`
+     * method to process received message. If the `retryable` processor is unsuccessful with a `retryable` error, after
+     * maximum numbers of attempts allowed, the message is republished to `-retry` topic for failover processing.
```

Comment:
This should be `If the `retryable` processor is unsuccessful with a retryable error, after maximum numbers of attempts allowed, the message is republished to -error topic for failover processing` (wrong topic).

Reply:
When the app is running in error mode, does the `-error` consumer not retry a `retryable` error 3 times, followed by publishing the same message to `-retry` topic, if all attempts are unsuccessful?

Reply:
The diagrams here may be helpful? https://companieshouse.atlassian.net/wiki/spaces/DEV/pages/129564747/Kafka+resilience#Kafkaresilience-HighLevelOverview

Reply:
The initial implementation was based on that diagram, where a message failing (with retryable error) on `-main` consumer would publish the message on to `-retry` topic and the `-retry` topic would do the same and publish it to `-error` topic, which would, in turn, publish it back to `-retry` if it also failed (with retryable error).

Reply:
I don't think that diagram shows any (automated) recovery/retry from the error topic does it?

Reply:
Yes, the `-error` to `-retry` publishing of failed message is listed under point `3a` of **Error Consumer** section of the story GCI-1067.